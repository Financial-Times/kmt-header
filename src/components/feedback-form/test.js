import React from 'react';
import { mount } from 'enzyme';
import FeedbackForm from './index.js';

describe("<FeedbackForm />", () => {
	const mockDispatch = jest.fn();

	const feedbackFormComponent = mount(
		<FeedbackForm dispatch={mockDispatch} isValid={false} />
	);

	it('should render without errors', () => {
	});

	describe('comment field', () => {
		const textareaFields = feedbackFormComponent.find('textarea');

		it('should appear once on the form', () => {
			expect(textareaFields.length).toBe(1);
		});

		it('should have its own label', () => {
			expect(feedbackFormComponent.find(`label[htmlFor="${textareaFields.prop('id')}"]`).length).toBe(1);
		});
	});

	describe('feedback score', () => {
		const scoreOptions = feedbackFormComponent.find('[name="npsscore"]');
		
		it('should contain 5 options', () => {
			expect(scoreOptions.length).toBe(5);
		});

		it('should have a label for each option', () => {
			expect(feedbackFormComponent.find('.nps-label').length).toBe(5);

			expect(scoreOptions.everyWhere(scoreOptionField => {
				return feedbackFormComponent.find(`label[htmlFor="${scoreOptionField.prop('id')}"]`).length === 1;
			})).toBe(true);
		});
	});

	// it('has the right content', () => {
	// 	const feedbackFormComponentMounted = mount(
	// 		<FeedbackForm dispatch={mockDispatch} isValid={true} />
	// 	);

	// 	const content = 
	// 	// <div>
 //  //       <form method="POST" className="kat-feedback" action="#">
 //          <div className="kat-feedback__row">
 //            <p className="kat-feedback__intro-text">Please share your feedback on the new Knowledge and Administration Tool (KAT) so that we can continue to develop it in line with customer requirements.</p>
 //          </div>

	// 	expect(feedbackFormComponentMounted.contains(content)).toBe(true);
	// });
	
	describe('submit button', () => {
		const submitButton = feedbackFormComponent.find('.kat-feedback__submit');

		it('should be enabled when the form is valid', () => {
			feedbackFormComponent.setProps({isValid: true});
			expect(submitButton.is('[disabled]')).toBe(false);
		});

		it('should be disabled when the form is not valid', () => {
			feedbackFormComponent.setProps({isValid: false});
			expect(submitButton.is('[disabled]')).toBe(true);
		});
	});
});
