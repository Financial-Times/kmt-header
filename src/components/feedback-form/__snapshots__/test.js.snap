// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FeedbackForm /> has the right content 1`] = `
"<div>
<form method=\\"POST\\" className=\\"kat-feedback\\" action=\\"#\\">
<div className=\\"kat-feedback__row\\">
<p className=\\"kat-feedback__intro-text\\">
Please share your feedback on the new Knowledge &amp; administration tools (KAT) so that we can continue to develop it in line with customer requirements.
</p>
</div>
<div className=\\"kat-feedback__row\\">
<label className=\\"o-forms__label\\">
How satisfied are you with FT KAT?
</label>
<div className=\\"kat-feedback__score\\">
<div>
<input type=\\"radio\\" name=\\"npsscore\\" value=\\"5\\" className=\\"o-forms__radio o-forms__radio--small\\" id=\\"nps5\\" onChange={[Function]} />
<label htmlFor=\\"nps5\\" className=\\"o-forms__label nps-label\\">
5 - Extremely satisfied
</label>
</div>
<div>
<input type=\\"radio\\" name=\\"npsscore\\" value=\\"4\\" className=\\"o-forms__radio o-forms__radio--small\\" id=\\"nps4\\" onChange={[Function]} />
<label htmlFor=\\"nps4\\" className=\\"o-forms__label nps-label\\">
4 - Somewhat satisfied
</label>
</div>
<div>
<input type=\\"radio\\" name=\\"npsscore\\" value=\\"3\\" className=\\"o-forms__radio o-forms__radio--small\\" id=\\"nps3\\" onChange={[Function]} />
<label htmlFor=\\"nps3\\" className=\\"o-forms__label nps-label\\">
3 - Neither satisfied nor dissatisfied
</label>
</div>
<div>
<input type=\\"radio\\" name=\\"npsscore\\" value=\\"2\\" className=\\"o-forms__radio o-forms__radio--small\\" id=\\"nps2\\" onChange={[Function]} />
<label htmlFor=\\"nps2\\" className=\\"o-forms__label nps-label\\">
2 - Somewhat dissatisfied
</label>
</div>
<div>
<input type=\\"radio\\" name=\\"npsscore\\" value=\\"1\\" className=\\"o-forms__radio o-forms__radio--small\\" id=\\"nps1\\" onChange={[Function]} />
<label htmlFor=\\"nps1\\" className=\\"o-forms__label nps-label\\">
1 - Extremely dissatisfied
</label>
</div>
</div>
</div>
<div className=\\"kat-feedback__row\\">
<label className=\\"o-forms__label\\" htmlFor=\\"improvefeedback\\">
How can we improve KAT?
</label>
<textarea className=\\"o-forms__textarea kat-feedback__textarea\\" name=\\"improvefeedback\\" id=\\"improvefeedback\\" onChange={[Function]} />
</div>
<div className=\\"kat-feedback__row\\">
<button className=\\"kat-feedback__submit\\" disabled={true} type=\\"button\\">
Submit
</button>
</div>
</form>
</div>"
`;
